manifests/init.pp

```diff
@@ -42,7 +53,9 @@
   $version               = 'present',
   $package               = undef,
   $java_alternative      = undef,
-  $java_alternative_path = undef
+  $java_alternative_path = undef,
+  $provider              = undef,
+  $source                = undef,
```

Comment:
It'd be great to have these parameters called `$package_...` to denote their connection to the package.